src/uiTest/scala/com/virtuslab/gitmachete/uitest/RunningIntelliJFixtureExtension.scala

```diff
@@ -84,26 +115,121 @@ trait RunningIntelliJFixtureExtension extends RobotPluginExtension { this: IdePr
         runJs("project.configure()")
       }
 
+      def moveMouseToTheMiddleAndWait(secondsToWait: Int): Unit = {
+        runJs(s"project.moveMouseToTheMiddleAndWait('$secondsToWait')");
+      }
+
+      def switchRepo(idx: Int): Unit = {
+        runJs(s"project.findComboBoxAndSwitchRepo('$idx')")
+      }
+
+      object contextMenu {
+
+        def openContextMenu(branch: String): Unit = {
+          runJs(s"project.findCellAndRightClick('$branch')")
+          probe.await()
+        }
+
+        def checkout(): Unit = {
+          runJs(s"project.contextMenu.checkout()")
+          probe.await()
+        }
+
+        def checkoutAndSyncByRebase(): Unit = {
+          runJs(s"project.contextMenu.checkoutAndSyncByRebase()")
+          // Consciously not calling probe.await() after triggering rebase action. The method freezes UI
+          // and makes impossible to proceed with actions (e.g. findAndClickButton('Start Rebasing')).
+          // This comment applies here, below and to project level counterparts.
+        }
+
+        def syncByRebase(): Unit = {
+          runJs(s"project.contextMenu.syncByRebase()")
+        }
+
+        def checkoutAndSyncByMerge(): Unit = {
+          runJs(s"project.contextMenu.checkoutAndSyncByMerge()")
+          probe.await()
+        }
+
+        def syncByMerge(): Unit = {
+          runJs(s"project.contextMenu.syncByMerge()")
+          probe.await()
+        }
+
+        def overrideForkPoint(): Unit = {
+          runJs(s"project.contextMenu.overrideForkPoint()")
+          probe.await()
+        }
+
+        def push(): Unit = {
+          runJs(s"project.contextMenu.push()")
+          probe.await()
+        }
+
+        def pull(): Unit = {
+          runJs(s"project.contextMenu.pull()")
+          probe.await()
+        }
+
+        def resetToRemote(): Unit = {
+          runJs(s"project.contextMenu.resetToRemote()")
+          probe.await()
+        }
+
+        def fastForwardMerge(): Unit = {
+          runJs(s"project.contextMenu.fastForwardMerge()")
+          probe.await()
+        }
+
+        def slideIn(): Unit = {
+          runJs(s"project.contextMenu.slideIn()")
+          probe.await()
+        }
+
+        def slideOut(): Unit = {
+          runJs(s"project.contextMenu.slideOut()")
+          probe.await()
+        }
+
+        def showInGitLog(): Unit = {
+          runJs(s"project.contextMenu.showInGitLog()")
+          probe.await()
+        }
+      }
+
+      object toolbar {
+
+        def toggleListingCommits(): Unit = {
+          runJs(s"project.toolbar.toggleListingCommits()")
+          probe.await()
+        }
+
+        def fetchAll(): Unit = {
+          runJs(s"project.toolbar.fetchAll()")
+          probe.await()
+        }
+      }
+
       def discoverBranchLayout(): Unit = {
         runJs("project.discoverBranchLayout()")
-        intelliJ.probe.await()
+        probe.await()
       }
 
       def fastForwardMergeCurrentBranchToParent(): Unit = {
```

Comment:
Should/can the existing methods also be moved under `contextMenu`/`toolbar` objects?

Reply:
Oh.. okay, makes sense, these methods kinda circumvent context menu/toolbar, instead calling the action directly hmm

Reply:
Would it make sense to open another issue to rewrite UITestSuite to these new indirect methods and remove stuff like `fastForwardMergeCurrentBranchToParent`? As your new methods seem superior wrt. coverage (maybe at the expense of slightly longer runtime?)

Reply:
`new methods seem superior` - yes. I think it may be worth to use them instead. 
Ofc timing can be an issue here.   

I will check that!